Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/ScrollingObjectCollection.cs

```diff
@@ -0,0 +1,2198 @@
+                    return ((int)Mathf.Floor((-(scrollContainer.transform.localPosition.x) + occlusionPositionPadding.x) / CellWidth) * Tiers) + (ViewableArea * Tiers) - 1;
+                }
+
+            }
+        }
+
+        //The empty game object that contains our nodes and be scrolled
+        [SerializeField]
+        [HideInInspector]
+        private GameObject scrollContainer;
+
+        //The empty game object that contains the ClipppingBox
+        [SerializeField]
+        [HideInInspector]
+        private GameObject clippingObject;
+
+        /// <summary>
+        /// The empty <see cref="GameObject"/> containing the <see cref="ScrollingObjectCollection"/>'s <see cref="ClippingBox"/>.
+        /// </summary>
+        public GameObject ClippingObject => clippingObject;
+
+        //A reference to the ClippingBox on the clippingObject
+        [SerializeField]
+        [HideInInspector]
+        private ClippingBox clipBox;
+
+        /// <summary>
+        /// The <see cref="ScrollingObjectCollection"/>'s <see cref="ClippingBox"/> that is used for clipping items in and out of the list.
+        /// </summary>
+        public ClippingBox ClipBox => clipBox;
+
+        //The bounds of the clipping object, this is to make helper math easier later, it doesn't matter that its AABB since we're really not using it for bounds operations
+        [SerializeField]
+        [HideInInspector]
+        private Bounds clippingBounds;
+
+        // Tracks whether an item in the list is being interacted with
+        private bool isEngaged = false;
+
+        // Tracks whether a movement action resulted in dragging the list  
+        private bool isDragging = false;
+
+        // we need to know if the pointer was a touch so we can do the threshold test (dot product test)
+        private bool isTouched = false;
+
+        //The postion of the scollContainer before we do any updating to it
+        private Vector3 initialScrollerPos;
+
+        //The new of the scollContainer before we've set the postion / finished the updateloop
+        private Vector3 workingScrollerPos;
+
+        //The node in the list currently being interacted with
+        private GameObject focusedObject;
+
+        //A list of new child nodes that have new child renderers that need to be added to the clippingBox
+        private List<ObjectCollectionNode> nodesToClip = new List<ObjectCollectionNode>();
+
+        //A list of new child nodes that have new child renderers that need to be removed to the clippingBox
+        private List<ObjectCollectionNode> nodesToUnclip = new List<ObjectCollectionNode>();
+
+        #region scroll state variables
+
+        // The currently tracked finger/pointer objects
+        private Transform currentPointerTransform;
+
+        private Handedness currentHand;
+
+        private IMixedRealityPointer currentPointer;
+
+        //The point where the original PointerDown occured
+        private Vector3 pointerHitPoint;
```

Comment:
There's no need to store this, since you are already storing currentPointer. Will help with readability of the code (there are a lot of fields...) to reduce unneeded fields.